### PR TITLE
Integrating sigclient in siglens

### LINKED ISSUE #269: [STORY] Move sigclient repo into this repo
# Description
We have a test tool that sits in https://github.com/sigscalr/sigscalr-client, move all of the files from this repo into this repo at location `tools/sigclient`

# Acceptance Criteria
1. The tool files are moved in its own dir
2. the client's readme files contains some cmds to run the load, make sure those cmds continue to work

### PR DESCRIPTION
**Pull request by darshannere**
# Description
This pull request addresses the task of relocating the Sigscalr Client Test Tool from its current location in the https://github.com/sigscalr/sigscalr-client repository to a new dedicated directory at tools/sigclient. The goal is to organize the codebase and make it more manageable.

Fixes #269  (link all the GitHub issues this addresses) creating a second cleaner Pull request

# Changes Made

1. Directory Structure:
- Moved all Sigscalr Client Test Tool files to the new directory: tools/sigclient.
- Ensured that the tool files are now contained within their own dedicated directory.

2. Readme Updates:
- Updated the client's readme file to reflect the changes in the tool's location.
- Verified that the readme file contains accurate and updated commands to run the load.

# Testing

1. Functional Testing:
- Confirmed that the Sigscalr Client Test Tool functions correctly in its new directory.
- Executed relevant commands to ensure that the load can still be run seamlessly.
2. Documentation Verification:
- Reviewed and verified the accuracy of the instructions in the client's readme file.
- Ensured that any references to the tool's location or execution commands were updated.

# Checklist:

- [x] I have self-reviewed this PR.
- [x] I have removed all print-debugging and commented-out code that should not be merged.
- [x] I have added sufficient comments in my code, particularly in hard-to-understand areas.
- [x] I have formatted the code, if applicable. For Go, I have run `goimports -w .`.

### CODE REVIEW

**Review by nbhavana — COMMENTED** [review]
Everything looks good so far. Only one thing, you are missing the   the `ingest.go` in the pkg folder.

**Comment by darshannere** [comment]
I will add the `ingest.go` file

**Comment by darshannere** [comment]
Hey, I have added the ingest.go file It was giving a problem because gitignore file of siglens had ingest*/

File: .gitignore [hunk 1]
```diff
@@ -79,5 +79,5 @@ node_modules
 siglens
 
 # multinode testing
-ingest*/
+#ingest*/
```
Comment (AndrewHess) [inline .gitignore:82]: If we don't want this, remove it instead of commenting it out

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -91,14 +94,18 @@ You should be able to access `http://localhost:5122` and see the SigLens UI. If
...
 ### Send Data to SigLens
 
-To send data, clone the [sigscalr-client](https://github.com/sigscalr/sigscalr-client) repo.
+To send data, go to `tools/sigclient`.
 
-In another terminal, start the ingestion via `sigscalr-client` by running:
+In another terminal, go to the location of sigclient:
+```
+cd tools/sigclient
+```
```
Comment (AndrewHess) [inline CONTRIBUTING.md:102]: You can delete line 97 since you have the same instruction in line 101

**Review by AndrewHess — COMMENTED** [review]
This looks pretty good

@nkunal do we need the tools/sigclient/.gitignore or the tools/sigclient/.github/workflows files (lint.yml and publish-prod-image.yml)? Also, are we going to delete the sigscalr-client repo after merging this PR? Alternatively, should we use a `git submodule` to incorporate the client into siglens instead of copying the files into siglens?

**Comment by nkunal** [comment]
> This looks pretty good
> 
> @nkunal do we need the tools/sigclient/.gitignore or the tools/sigclient/.github/workflows files (lint.yml and publish-prod-image.yml)? Also, are we going to delete the sigscalr-client repo after merging this PR? Alternatively, should we use a `git submodule` to incorporate the client into siglens instead of copying the files into siglens?

we dont need the gitignore file and the workflow files, you can remove them from here.

We will delete the sigscalr-client repo after merging this PR. So we don't need the git submodule

File: .gitignore (hunk 1 above)
Comment (darshannere) [reply .gitignore:82]: Yes, Will update it

File: CONTRIBUTING.md (hunk 2 above)
Comment (darshannere) [reply CONTRIBUTING.md:102]: sure

**Comment by darshannere** [comment]
Removed the two suggested lines

**Comment by AndrewHess** [comment]
@darshannere I'm still seeing the `tools/sigclient/.github/workflows` directory and the `tools/sigclient/.gitignore` file in this PR. Can you remove these?

Also the siglens-lint-ut test is failing because the Go code isn't properly formatted. You can automatically format the current directory with `goimports -w .`

**Pull request closed without merging**